pkgs/by-name/wr/wrangler/package.nix

```diff
@@ -0,0 +1,110 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+  fetchFromGitHub,
+  pnpm_9,
+  nodejs,
+  npmHooks,
+  writeScriptBin
+}:
+let
+  srcHash = "sha256-/4iIkvSn85fkRggmIha2kRlW0MEwvzy0ZAmIb8+LpZQ=";
+  pnpmDepsHash = "sha256-aTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=";
+in
+  stdenv.mkDerivation (finalAttrs: {
+  pname = "wrangler";
+  version = "3.62.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "workers-sdk";
+    rev = "wrangler@${finalAttrs.version}";
+    hash = "${srcHash}";
+  };
+
+  buildInputs = [
+    pkgs.llvmPackages.libcxx
+    pkgs.llvmPackages.libunwind
+    pkgs.musl
+    pkgs.xorg.libX11
+  ] ++ lib.optional stdenv.isLinux pkgs.autoPatchelfHook;
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm_9.configHook
+    pkgs.autoPatchelfHook
+  ];
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "${pnpmDepsHash}";
+  };
+
+  preBuild = ''
+    addAutoPatchelfSearchPath lib/node_modules/.pnpm/@cloudflare+workerd-linux-64@1.20240620.1/node_modules/@cloudflare/workerd-linux-64/bin/
+    '';
+
+  # @cloudflare/vitest-pool-workers wanted to run a server as part of the build process
+  # so I simply removed it
+  postBuild = ''
+    rm -fr packages/vitest-pool-workers
+    NODE_ENV="production" pnpm --filter miniflare run build
+    NODE_ENV="production" pnpm --filter wrangler run build
+    # pnpm --offline deploy --frozen-lockfile  --ignore-script  --filter=bash-language-server server-deploy
```

Comment:
This line doesn't need to be here, right?

Reply:
It definitely doesn't, I think it's there because that is what the normal example is for what a standard pnpm nixos build "should" look like. This should be removed before merging but also we need to figure out if what we have above this line is actually "good enough".